### PR TITLE
Re-implement some changes that were dropped when #5042 was merged

### PR DESCRIPTION
**Pull request by adamrtalbot**
Some changes were dropped when @swampie's PR [was merged](https://github.com/nextflow-io/nextflow/pull/4897). This PR re-adds them from scratch, will try and trim it down so we retain the relevant lines.

### CODE REVIEW

**Comment by pditommaso** [comment]
Can be added an integration tests here 

https://github.com/nextflow-io/nextflow/blob/658566e83a009c36be3591ac077f650bc1eefc95/validation/azure.sh#L57-L65

File: plugins/nf-azure/src/main/nextflow/cloud/azure/batch/AzBatchService.groovy [hunk 1]
```diff
@@ -299,9 +299,14 @@ class AzBatchService implements Closeable {
...
 
-        return new ManagedIdentityCredentialBuilder()
-                .clientId(config.managedIdentity().clientId)
-                .build()
+        final clientId = config.managedIdentity().clientId
+        final credential = new ManagedIdentityCredentialBuilder()
+
+        if (clientId)
+            log.debug "[AZURE BATCH] Managed Identity Client ID: ${clientId}"
+            credential.clientId(clientId)
```
Comment (bentsherman) [inline plugins/nf-azure/src/main/nextflow/cloud/azure/batch/AzBatchService.groovy:307]:
```suggestion
        if( clientId )
            credential.clientId(clientId)
```

Comment (pditommaso) [reply plugins/nf-azure/src/main/nextflow/cloud/azure/batch/AzBatchService.groovy:307]: well spotted.

Comment (pditommaso) [reply plugins/nf-azure/src/main/nextflow/cloud/azure/batch/AzBatchService.groovy:307]: there's a dangerous tendency to write code without unit tests, tho

**Comment by pditommaso** [comment]
Merge manually 306814e7

**Pull request closed without merging**